core/src/main/java/org/fao/geonet/kernel/search/log/QueryRequest.java

```diff
@@ -282,47 +287,116 @@ public boolean isSimpleQuery() {
+//                    Log.debug(Geonet.SEARCH_LOGGER, "No queryInfo objects to insert into database for this query: " + this.luceneQuery);
+//                }
+//            }
+//        }
+//
+//        requestRepo.save(request);
+//        if (Log.isDebugEnabled(Geonet.SEARCH_LOGGER))
+//            Log.debug(Geonet.SEARCH_LOGGER, "QueryRequest inserted (id: " + this.requestId + ")");
+//        return true;
+//    }
+
+
+    @Value("${es.index.searchlogs}")
+    private String index = "searchlogs";
+
+    @Autowired
+    private EsClient client;
+
+    public boolean storeToEs() {
+        Map<String, String> listOfDocumentsToIndex = new HashMap<>();
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            String id = UUID.randomUUID().toString();
+
+            ObjectNode doc = mapper.createObjectNode();
+            final long time = this.date.getTime();
+            doc.put("id", id);
+            doc.put("documentType", "request");
+            doc.put("timestamp", time);
+            doc.put("host", this.ip);
+            doc.put("path", this.service);
+            doc.put("hits", this.hits);
+            doc.put("language", this.language);
+            doc.put("query", this.luceneQuery);
+            doc.put("recordType", this.mdType.name());
+            doc.put("sortBy", this.sortBy);
+            doc.put("spatialFilter", this.spatialFilter);
+
+            listOfDocumentsToIndex.put(id, mapper.writeValueAsString(doc));
+
+
+
+            final List<SearchRequestParam> queryInfos = this.queryInfos;
+            final int[] i = {0};
+            queryInfos.forEach(q -> {
+                ObjectNode paramsDoc = mapper.createObjectNode();
+                String type = "searchParam";
+                i[0]++;
+                String pId = id + type + i[0];
+                paramsDoc.put("id", pId);
+                paramsDoc.put("documentType", type);
+                paramsDoc.put("timestamp", time);
+                paramsDoc.put("field", q.getTermField());
+                paramsDoc.put("fieldValue", q.getTermText());
+                paramsDoc.put("similarity", q.getSimilarity());
+                paramsDoc.put("fieldLowerValue", q.getLowerText());
+                paramsDoc.put("fieldUpperValue", q.getUpperText());
+                paramsDoc.put("inclusive", q.isInclusive());
+                try {
+                    listOfDocumentsToIndex.put(pId, mapper.writeValueAsString(paramsDoc));
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
                 }
-            }
+            });
+
+            listOfDocumentsToIndex.put(id, mapper.writeValueAsString(doc));
+
+            client = ApplicationContextHolder.get().getBean(EsClient.class);
+            client.bulkRequest(index, listOfDocumentsToIndex);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return false;
+        } catch (IOException e) {
```

Comment:
Only `JsonProcessingException` and `IOException` are catch. Can happen other exceptions and if so should be catched to avoid if any error in ES indexing affects the current feature in GeoNetwork?